File: 09.Greed贪心/leetcode_3689_maxTotalValue.py
from typing import List

class Solution:
    def maxTotalValue(self, nums: List[int], k: int) -> int:
        return k*(max(nums)-min(nums))

if __name__ == "__main__":
    s = Solution()
    nums = [1,3,2]
    k = 2
    print(s.maxTotalValue(nums, k))

File: 09.Greed贪心/leetcode_3689_maxTotalValue.cpp
#include <iostream>
#include <vector>
#include <climits>
using namespace std;

class Solution {
public:
    long long maxTotalValue(vector<int>& nums, int k) {
        long long mn = INT_MAX, mx = 0;
        for (int i : nums){
            mn = min(mn, 1ll*i);
            mx = max(mx, 1ll*i);
        }
        return 1ll*k*(mx-mn);
    }
};

int main(){
    Solution s;
    vector<int> nums = {1,3,2};
    int k = 2;
    cout << s.maxTotalValue(nums, k) << endl;
}
